File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n)
    {
        string x="1";
        
        for(int i=2;i<=n;i++)
        {string ans="";
            int l=x.size();
            char c= x[0];
            int counter=1;
            for(int j=1;j<l;j++)
            {
               if(c==x[j])
               {
                   counter++;
               }else
               {
                   ans+=to_string(counter)+c;
                   c=x[j];
                   counter=1;
               } 
            }if(counter>0)
            {ans+=to_string(counter)+c;
                
            }
            
            
        x=ans;
        }    
      return x;  
    }
};

File: 0038-count-and-say/0038-count-and-say.java
class Solution {
    String solve(int n,String s){
        if(n==1) return s;
        int ns=s.length();
        StringBuilder sbr=new StringBuilder();
        int cnt=1;
        for(int i=1;i<ns;i++){
            if(s.charAt(i)==s.charAt(i-1)){
                cnt++;
            }else{
                sbr.append(String.valueOf(cnt));
                sbr.append(s.charAt(i-1));
                //sbr+=s.charAt(i-1);
                cnt=1;
            }
        }
        sbr.append(String.valueOf(cnt));
        sbr.append(s.charAt(ns-1));
        return solve(n-1,sbr.toString());


    }
    public String countAndSay(int n) {
        return solve(n,"1");
    }
}